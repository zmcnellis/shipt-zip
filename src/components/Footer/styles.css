/* Footer --------------------------------------------------------------------*/
footer {
  display: table;
  text-align: center;
  background-color: #414042;
  width: 100%;
  height: 75px;
  color: #f7f4ea;
  margin-top: 40px;
}

.footer-text {
  display: table-cell;
  vertical-align: middle;
  font-size: 11px;
}
