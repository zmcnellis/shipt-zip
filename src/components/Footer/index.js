import React, { Component } from 'react';
import './styles.css';

class Footer extends Component {
  // because there can only be one footer and it doesn't typically change, the text is set here instead of in props
  render() {
    return (
      <footer>
        <p className="footer-text">&copy; 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.</p>
      </footer>
    );
  }
}

export default Footer;
