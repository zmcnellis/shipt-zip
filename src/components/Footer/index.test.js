import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Footer from '.';

describe('Footer', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Footer/>
    );
    expect(output).toMatchSnapshot();
  });

  it('should display the correct text', () => {
    const output = shallow(
      <Footer/>
    );
    expect(output.find('p').hasClass('footer-text')).toEqual(true);
    expect(output.text()).toEqual('Â© 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.');
  });
});
