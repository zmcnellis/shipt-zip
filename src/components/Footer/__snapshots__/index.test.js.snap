// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="footer-text"
>
        © 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "© 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.",
        "className": "footer-text",
      },
      "ref": null,
      "rendered": "© 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.",
      "type": "p",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="footer-text"
>
          © 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.",
          "className": "footer-text",
        },
        "ref": null,
        "rendered": "© 2018 Shipt, Inc. and its services are not necessarily affiliated or endorsed by the retailers on this site.",
        "type": "p",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
